File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/managetaskview/ITaskViewContract.java
package com.wiseassblog.samsaradayplanner.ui.managetaskview;

import com.wiseassblog.samsaradayplanner.domain.Task;
import com.wiseassblog.samsaradayplanner.domain.constants.COLOR;
import com.wiseassblog.samsaradayplanner.domain.constants.ICON;

public interface ITaskViewContract {
    interface View {
        void showColorPickerSheet();
        void hideColorPickerSheet();
        void setName(String name);
        String getName();
        void setIconSelection(ICON icon);
        void setSelectedSpinnerItem(int position);
        void setButtonColor(COLOR c);
        void goToTaskListActivity();
        void showMessage(String message);
    }

    interface ViewModel {
        void setTask(Task task);
        Task getTask();
    }
}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/managetaskview/ITaskViewContract.kt
package com.wiseassblog.samsaradayplanner.ui.managetaskview

import com.wiseassblog.samsaradayplanner.domain.Task
import com.wiseassblog.samsaradayplanner.domain.constants.ICON
import com.wiseassblog.samsaradayplanner.domain.constants.COLOR

interface ITaskViewContract {
    interface View {
        fun showColorPickerSheet()
        fun hideColorPickerSheet()
        var name: String
        fun setIconSelection(icon: ICON)
        fun setSelectedSpinnerItem(position: Int)
        fun setButtonColor(c: COLOR)
        fun goToTaskListActivity()
        fun showMessage(message: String)
    }

    interface ViewModel {
        var task: Task
    }
}